### ISSUE TITLE
Ensure that ICD10 mappings are ICD10CM, not ICD10WHO.

### ISSUE THREAD

**Issue by matentzn** [monarch-initiative/mondo#4651]
We so far made the flawed assumption that and an id with the ICD10 prefix is in fact ICD10CM if only the id is a legal ICD10 CM code. However, there is no guarantee this is enough. According to @drseb there could be differences between the exact same code in ICD10WHO and ICD10CM. So we need to make sure that all our codes are not only syntactically legal, but also correct. I do not know how to solve this yet, so it needs some thought. Hopefully this problem is not that huge.

**Comment by matentzn** [comment]
Also, I am really hoping that codes do not _change_ between year versions of ICD10CM, in the sense that a code S73.0 means the same thing in 2016 as in 2019.

**Comment by matentzn** [comment]
Chris C.: "I am 99% certain that ICD10 and ICD10CM preserve 4-digit fidelity.  After four digits, all bets are off."

**Comment by matentzn** [comment]
Interesting but mostly unrelated to this issue here: 
- https://www.aapc.com/icd-10/conversion-mapping.aspx
- https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6911227/
-

**Comment by drseb** [comment]
@matentzn can you send chris c the 4-digit code I sent you?

**Comment by matentzn** [comment]
- https://icd.who.int/browse10/2019/en#/S73.0 Dislocation of Hip
- https://www.icd10data.com/ICD10CM/Codes/S00-T88/S70-S79/S73-/S73.0 Subluxation and dislocation of hip

@drseb can you describe you worry? Do you feel the mention of `Subluxation` changes the clinical meaning of the code?

**Comment by matentzn** [comment]
Idea for strategy:

- We should build a simple ICD 10 WHO ingest: https://github.com/monarch-initiative/mondo-ingest/issues/8
- All codes in ICD10CM that are not in ICD10WHO are safe
- All codes in ICD10CM that have the same label as ICD10WHO are safe
- All codes in ICD10CM that have a different label than ICD10WHO are potentially dangerous
- All codes in ICD10CM that have a correspond to a _different_ ICD10WHO code with the same label are high danger
- The latter two need to be put in a table side by side:

| mondo_id | mondo_label | icd10cm_id | icd10cm_label | icd10who_id | icd10who_label |
| ---------- | ------------- | ----------- | --------------- | ------------ | ---------------- |
| MONDO:123 | Hip dislocation | S73.0 | Dislocation of Hip | S73.0 | Subluxation and dislocation of hip | 

We then need to review these manually (hopefully they are not too many)

**Comment by joeflack4** [comment]
This sounds good.

I went ahead and numbered the bullet points in your previous comment. In regards to (3) and (4), I think it is prudent to split the criteria into two: (i) `bool`: exact matches / non-matches, (ii) `float [0-1]`: string similarity, along with thresholds.

**Comment by joeflack4** [comment]
Fuzzy matching options (this is not my area of expertise, yet):
a. Semantic similarity: Compare the words in a given term with where they stand in relation to the whole code system.
b. Word embeddings: Vectorize the corpus. 
c. Graph embeddings:

**Comment by sabrinatoro** [comment]
We compared Mondo code vs ICD10CM vs ICD10WHO, and determined which one of ICD10CM or ICD10WHO matches Mondo the best. See [this spreadsheet for details](https://docs.google.com/spreadsheets/d/1o4__JLNKIonxUN4Yc8Kunf5sA_Nrscg1ZTuVF4z8v7M/edit#gid=45343105).

- total Mondo terms: 1099
- Mondo terms for which ICD10CM and ICD10WHO are equivalent: 858 
  - the ICD code have either the same labels, or have similar labels (e.g. "hepatic disease" vs "liver disease")
  - the terms labels are used to represent the same concepts 
- Mondo terms for which ICD10CM is the best match : 222
  - 215 : do not have ICD10WHO equivalent
  - 7 : ICD10WHO code seems narrower or broader than ICD10CM code, and the ICD10CM code matched better the Mondo code. 
- Mondo terms for which ICD10WHO is the best match: 18
  - 7: do not have ICD10CM equivalent
  - 11: ICD10CM code seems narrower or broader than ICD10WHO code, and the ICD10WHO code matched better the Mondo code. 

In summary: it looks like Mondo mappings were originally made to ICD10CM.

@matentzn

**Comment by joeflack4** [comment]
Good to have these results! So it seems that, generally, both ICD10 flavors are closely related, and where they differ currently, the most of the mondo mappings correspond to ICD10CM.

**Comment by matentzn** [comment]
@sabrinatoro can you for the 18 codes for which ICD10WHO code is a better match, change the Prefix in the xref to ICD10WHO and assign me as a reviewer please?

This also closes this issue for good (i..e you can add `fixes #4651` to the pull request).

Thank you all so much on your great work on this diversion here, esp @joeflack4 and @sabrinatoro!

**Comment by sabrinatoro** [comment]
- when there was no ICD10CM, I updated the db-xref to ICD10WHO
- when there was a ICD10CM, I added the ICD10WHO and reviewed the ICD10CM. I most (almost all) cases, the ICD10CM was still an acceptable equivalent, even though the label seemed closer in ICD10WHO.

**Comment by matentzn** [comment]
Great!

**Comment by joeflack4** [comment]
@sabrinatoro @matentzn On Friday you asked if I could create a `icd10cm-icd10who_similarity` column. I've done that, and also updated [the google sheet](https://docs.google.com/spreadsheets/d/1o4__JLNKIonxUN4Yc8Kunf5sA_Nrscg1ZTuVF4z8v7M/edit#gid=45343105
[mondo_icd10_xref_analysis.csv](https://github.com/monarch-initiative/mondo/files/8484059/mondo_icd10_xref_analysis.csv)
).
